apps/src/codebridge/hooks/useSource.ts

```diff
@@ -148,6 +149,14 @@ export const useSource = (defaultSources: ProjectSources) => {
       initialSources !== previousInitialSources.current
     ) {
       if (initialSources) {
+        // Set the last source in project manager to initial sources.
+        // This prevents us from immediately saving the source on load,
+        // as we only want to save when the user makes a change.
+        // Initial sources always comes from the server, so we never need to save
+        // it again.
+        Lab2Registry.getInstance()
+          .getProjectManager()
+          ?.setLastSource(initialSources);
```

Comment:
Should we just do this for all labs (i.e. in lab2Redux when we load all the level data)?

Reply:
The reason I didn't do it there is we do some pre-processing of the start sources [here](https://github.com/code-dot-org/code-dot-org/blob/6acb2919fa24a548ad1363f94571491dc90e044e/apps/src/codebridge/hooks/useInitialSources.ts#L75-L93), so the start code from the server doesn't match what we first try to save. I think you also do some pre-processing in Music Lab to add the library?

Reply:
Ah yeah fair - and yeah in that case makes sense for labs to handle if/how they want to save or not save after start sources have loaded